modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStore.java

```diff
@@ -17,122 +17,108 @@
  */
 public class FilePageStore implements PageStore {
-    /** File version. */
-    public static final int VERSION_1 = 1;
+    private static final VarHandle PAGE_COUNT;
 
-    /** Skip CRC calculation flag. */
-    // TODO: IGNITE-17011 Move to config
-    private final boolean skipCrc = getBoolean("IGNITE_PDS_SKIP_CRC");
+    private static final VarHandle NEW_DELTA_FILE_PAGE_STORE_IO_FUTURE;
 
-    /** File page store version. */
-    private final int version;
+    static {
+        try {
+            PAGE_COUNT = MethodHandles.lookup().findVarHandle(FilePageStore.class, "pageCount", int.class);
 
-    /** Page size in bytes. */
-    private final int pageSize;
+            NEW_DELTA_FILE_PAGE_STORE_IO_FUTURE = MethodHandles.lookup().findVarHandle(
+                    FilePageStore.class,
+                    "newDeltaFilePageStoreIoFuture",
+                    CompletableFuture.class
+            );
+        } catch (ReflectiveOperationException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
 
-    /** Header size in bytes. Should be aligned to the {@link #pageSize}. */
-    private final int headerSize;
+    /** File page store version. */
+    public static final int VERSION_1 = 1;
 
-    /** File page store path. */
-    private final Path filePath;
+    /** Delta file page store IO version. */
+    public static final int DELTA_FILE_VERSION_1 = 1;
 
-    /** {@link FileIo} factory. */
-    private final FileIoFactory ioFactory;
+    /** File page store IO. */
+    private final FilePageStoreIo filePageStoreIo;
 
     /** Page count. */
-    private final AtomicInteger pageCount = new AtomicInteger();
+    private volatile int pageCount;
 
-    private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+    /** New page allocation listener. */
+    private volatile @Nullable PageAllocationListener pageAllocationListener;
 
-    /** Caches the existence state of storage file. After it is initialized, it will be not {@code null} during lifecycle. */
-    private volatile Boolean fileExists;
+    /** Delta file page store IOs. */
+    private final List<DeltaFilePageStoreIo> deltaFilePageStoreIos;
 
-    /** {@link FileIo} for read/write operations with file. */
-    private volatile FileIo fileIo;
+    /** Future with a new delta file page store. */
+    private volatile @Nullable CompletableFuture<DeltaFilePageStoreIo> newDeltaFilePageStoreIoFuture;
 
-    /** Initialized file page store. */
-    private volatile boolean initialized;
+    /** {@link DeltaFilePageStoreIo} factory. */
+    private volatile @Nullable DeltaFilePageStoreIoFactory deltaFilePageStoreIoFactory;
 
-    /** New page allocation listener. */
-    private volatile @Nullable PageAllocationListener pageAllocationListener;
+    /** Callback on completion of delta file page store creation. */
+    private volatile @Nullable CompleteCreationDeltaFilePageStoreIoCallback completeCreationDeltaFilePageStoreIoCallback;
```

Comment:
Can't this thing just be a `newDeltaFilePageStoreIoFuture` listener? Is there a necessity for a specific field and a specific functional interface? Looks over-engineered, but maybe I'm wrong

Reply:
We personally decided that I would delete it.

Reply:
fix it